conwayconsole/Program.cs

```diff
@@ -9,140 +9,189 @@
+                      board = game.Turn(out stop))
+                    {
+                        if (!options.HideDisplay)
+                        {
+                            await Console.Out.WriteLineAsync(board.ToConsoleString(window, builder, getValueForColour));
+                            DateTime now = DateTime.UtcNow;
+                            TimeSpan elapsed = now.Subtract(lastLoopTime);
+                            TimeSpan delay = TimeSpan.FromMilliseconds(options.Delay).Subtract(elapsed);
+
+                            await Console.Out.WriteLineAsync($"{BoardConsoleExtensions.cHome}({board.Width}x{board.Height}) ({window.Width}x{window.Height}) ({window.X}, {window.Y})");
+                            if (delay > TimeSpan.Zero)
+                            {
+                                if (delay > TimeSpan.Zero)
+                                {
+                                    await Task.Delay(delay, cts.Token);
+                                }
+
+                                lastLoopTime = DateTime.UtcNow;
+                            }
+                        }
+
+                        if (save)
+                        {
+                            await Save(board);
+                            save = false;
+                            break;
+                        }
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                }
+                finally
+                {
+                    if (!options.HideDisplay)
+                    {
+                        Console.CursorVisible = true;
+                        Console.ResetColor();
+                        Console.CancelKeyPress -= HandleCancel;
+                    }
+                }
+
+                void HandleCancel(object sender, ConsoleCancelEventArgs args) => cts.Cancel();
             }
-          }
         }
-        catch (OperationCanceledException)
+
+        private static async Task<bool> YesNo(string prompt = null)
         {
+          await Console.Out.WriteAsync(prompt);
+          string response = (await Console.In.ReadLineAsync()).ToLower();
+          return response == "y" || response == "yes";
         }
-        finally
-        {       
-          if (!options.HideDisplay)
-          { 
-            Console.CursorVisible = true;
-            Console.ResetColor();
-            Console.CancelKeyPress -= HandleCancel;
-          }
-        }
+        private static async Task Save(IReadableBoard boardToSave)
+        {
+            if (boardToSave != null)
+            {
+                await Console.Out.WriteAsync("Enter file path (leave blank for default): ");
+                string pathway = await Console.In.ReadLineAsync();
 
-        void HandleCancel(object sender, ConsoleCancelEventArgs args) => cts.Cancel();
-      }
+                bool pathwayIsNullOrEmpty = string.IsNullOrEmpty(pathway);
```

Comment:
In the Issue, I suggested the use of `IsNullOrWhitespace` as it doesn't make sense to try to save a file whose name is something like "  "